runtime/src/accounts_db.rs

```diff
@@ -3089,56 +3131,148 @@ impl AccountsDb {
                 // confident that the entire state for this slot has been flushed to the storage
                 // already.
+                let mut scan_storages_elasped = Measure::start("scan_storages_elasped");
+                let scan_result: ScanStorageResult<Pubkey, DashSet<Pubkey>> = self
+                    .scan_account_storage(
+                        *remove_slot,
+                        |loaded_account: LoadedAccount| Some(*loaded_account.pubkey()),
+                        |accum: &DashSet<Pubkey>, loaded_account: LoadedAccount| {
+                            accum.insert(*loaded_account.pubkey());
+                        },
+                    );
+                scan_storages_elasped.stop();
+                purge_stats
+                    .scan_storages_elasped
+                    .fetch_add(scan_storages_elasped.as_us(), Ordering::Relaxed);
+
+                let mut purge_accounts_index_elapsed =
+                    Measure::start("purge_accounts_index_elapsed");
+                // Purge this slot from the accounts index
+                let purge_slot: HashSet<Slot> = vec![*remove_slot].into_iter().collect();
+                let mut reclaims = vec![];
+                match scan_result {
+                    ScanStorageResult::Cached(_) => {
+                        panic!("Should not see cached keys in this `else` branch, since we checked this slot did not exist in the cache above");
+                    }
+                    ScanStorageResult::Stored(stored_keys) => {
+                        for set_ref in stored_keys.iter() {
+                            self.accounts_index.purge_exact(
+                                set_ref.key(),
+                                &purge_slot,
+                                &mut reclaims,
+                            );
+                        }
+                    }
+                }
+                purge_accounts_index_elapsed.stop();
+                purge_stats
+                    .purge_accounts_index_elapsed
+                    .fetch_add(purge_accounts_index_elapsed.as_us(), Ordering::Relaxed);
+
+                // `handle_reclaims()` should remove all the account index entries and
+                // storage entries
+                let mut handle_reclaims_elapsed = Measure::start("handle_reclaims_elapsed");
+                // Slot should be dead after removing all its account entries
+                let expected_dead_slot = Some(*remove_slot);
+                self.handle_reclaims(
+                    &reclaims,
+                    expected_dead_slot,
+                    Some(purge_stats),
+                    Some(&mut ReclaimResult::default()),
+                    false,
+                );
+                handle_reclaims_elapsed.stop();
+                purge_stats
+                    .handle_reclaims_elapsed
+                    .fetch_add(handle_reclaims_elapsed.as_us(), Ordering::Relaxed);
```

Comment:
how about extracting this code into new fn like `purge_slot_storage()`. This is to align with `purge_slot_cache()` above.